clippy_lints/src/nonstandard_macro_braces.rs

```diff
@@ -93,13 +93,21 @@ impl EarlyLintPass for MacroBraces {
 }
 
 fn is_offending_macro<'a>(cx: &EarlyContext<'_>, span: Span, mac_braces: &'a MacroBraces) -> Option<MacroInfo<'a>> {
+    let unnested_or_local = || {
+        let nested = in_macro(span.ctxt().outer_expn_data().call_site);
+        let in_local_macro = nested
```

Comment:
Having `nested &&` here and returning `!nested || ..` might befuddle at cursory reading. I'd suggest using a guard `if` instead.

Reply:
Is this what you mean?
```rust
let unnested_or_local = || {
    let nested = in_macro(span.ctxt().outer_expn_data().call_site);
    let in_local_macro = matches!(
        span.macro_backtrace().last().and_then(|e| e.macro_def_id),
        Some(defid) if nested && defid.is_local() // adding `nested` to the if guard?
    );
    
    !nested || in_local_macro
};
```

Reply:
I was more thinking of

```
let unnested_or_local = || {
    let nested = in_macro(span.ctxt().outer_expn_data().call_site);
    !nested || span.macro_backtrace().last().map_or(false, |e| 
        e.macro_def_id.map_or(false, |defid| defid.is_local()))
    )
};
```

Reply:
Oh right duh :facepalm: I was asking the same question two times in a row (`!nested` then `nested && ..`)